debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbOffsetContext.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mongodb;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Struct;
+import org.bson.Document;
+
+import io.debezium.connector.SnapshotRecord;
+import io.debezium.pipeline.spi.OffsetContext;
+import io.debezium.pipeline.txmetadata.TransactionContext;
+import io.debezium.schema.DataCollectionId;
+
+/**
+ * A context that facilitates the management of the current offsets across a set of mongodb replica sets.
+ *
+ * @author Chris Cranford
+ */
+public class MongoDbOffsetContext implements OffsetContext {
```

Comment:
I wonder, does it make sense to have an interface hierarchy
SingleOfsetContext extends OffsetConext
PartitionedOffsetContext extends OffsetContext

Reply:
If I understand correctly, then mongo would use this `PartitionedOffsetContext` since it relies on multiple partitions per task instance where-as our other connectors would use the other implementation since they only rely on a single offset with a hard-coded partition?

Reply:
Added this as a follow-up in [DBZ-1860](https://issues.redhat.com/browse/DBZ-1860).